File: src/com/dafttech/terra/graphics/gui/anchors/AnchorMouse.java
package com.dafttech.terra.graphics.gui.anchors;

import com.badlogic.gdx.Gdx;
import com.dafttech.terra.graphics.gui.GUIObject;
import com.dafttech.terra.graphics.gui.containers.GUIContainer;

public class AnchorMouse extends GUIAnchor {
    @Override
    public void applyAnchor(GUIObject object, GUIContainer container) {
        object.position.x = Gdx.input.getX() + 10;
        object.position.y = Gdx.input.getY() - object.size.y - 10;

        if (object.position.x + object.size.x > Gdx.graphics.getWidth()) {
            object.position.x -= 20 + object.size.x;
        }

        if (object.position.y < 0) {
            object.position.y += object.size.y + 10;
        }
    }

    @Override
    public boolean needsApplyOnFrame() {
        return true;
    }
    
    @Override
    public boolean isContainerDependent() {
        return false;
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/anchors/AnchorMouse.scala
package com.dafttech.terra.engine.gui.anchors

import com.badlogic.gdx.Gdx
import com.dafttech.terra.engine.gui.GUIObject
import com.dafttech.terra.engine.gui.containers.GUIContainer
import com.dafttech.terra.engine.vector.Vector2d

class AnchorMouse extends GUIAnchor {
  def applyAnchor(gObj: GUIObject, container: GUIContainer): Unit = {
    gObj.position = Vector2d(Gdx.input.getX + 10, Gdx.input.getY - gObj.size.y - 10)
    if (gObj.position.x + gObj.size.x > Gdx.graphics.getWidth) {
      gObj.position = gObj.position.mapX(_ - 20 + gObj.size.x)
    }
    if (gObj.position.y < 0) {
      gObj.position = gObj.position.mapY(_ + gObj.size.y + 10)
    }
  }

  override def needsApplyOnFrame: Boolean = true

  override def isContainerDependent: Boolean = false
}